x-pack/plugins/infra/server/routes/snapshot/lib/trasform_metrics_ui_response.ts

```diff
@@ -47,42 +48,49 @@ export const transformMetricsApiResponseToSnapshotResponse = (
   source: InfraSource,
   metricsApiResponse: MetricsAPIResponse
 ): SnapshotNodeResponse => {
-  const nodes = metricsApiResponse.series.map((series) => {
-    const node = {
-      metrics: options.metrics
-        .filter((m) => m.id !== META_KEY)
-        .map((metric) => {
-          const name = metric.id as SnapshotMetricType;
-          const timeseries = {
-            id: name,
-            columns: [
-              { name: 'timestamp', type: 'date' as MetricsExplorerColumnType },
-              { name: 'metric_0', type: 'number' as MetricsExplorerColumnType },
-            ],
-            rows: series.rows.map((row) => {
-              return { timestamp: row.timestamp, metric_0: get(row, metric.id, null) };
-            }),
-          };
-          const maxValue = calculateMax(timeseries.rows);
-          const avg = calculateAvg(timeseries.rows);
-          const value = getLastValue(timeseries.rows);
-          const nodeMetric: SnapshotNodeMetric = { name, max: maxValue, value, avg };
-          if (snapshotRequest.includeTimeseries) {
-            nodeMetric.timeseries = timeseries;
-          }
-          return nodeMetric;
-        }),
-      path:
-        series.keys?.map((key) => {
-          return { value: key, label: key } as SnapshotNodePath;
-        }) ?? [],
-      name: '',
-    };
+  const nodes = metricsApiResponse.series
+    .map((series) => {
+      const node = {
+        metrics: options.metrics
+          .filter((m) => m.id !== META_KEY)
+          .map((metric) => {
+            const name = metric.id as SnapshotMetricType;
+            const timeseries = {
+              id: name,
+              columns: [
+                { name: 'timestamp', type: 'date' as MetricsExplorerColumnType },
+                { name: 'metric_0', type: 'number' as MetricsExplorerColumnType },
+              ],
+              rows: series.rows.map((row) => {
+                return { timestamp: row.timestamp, metric_0: get(row, metric.id, null) };
+              }),
+            };
+            const maxValue = calculateMax(timeseries.rows);
+            const avg = calculateAvg(timeseries.rows);
+            const value = getLastValue(timeseries.rows);
+            const nodeMetric: SnapshotNodeMetric = { name, max: maxValue, value, avg };
+            if (snapshotRequest.includeTimeseries) {
+              nodeMetric.timeseries = timeseries;
+            }
+            return nodeMetric;
+          }),
+        path:
+          series.keys?.map((key) => {
+            return { value: key, label: key } as SnapshotNodePath;
+          }) ?? [],
+        name: '',
+      };
 
-    const path = applyMetadataToLastPath(series, node, snapshotRequest, source);
-    const lastPath = last(path);
-    const name = (lastPath && lastPath.label) || 'N/A';
-    return { ...node, path, name };
-  });
+      const isNoData = node.metrics.every((m) => m.value === null);
+      const isAPMNode = series.metricsets?.includes('app');
```

Comment:
APM produces other metric sets than `app` so I'm not sure this check will filter all APM nodes out.
Perhaps it would be better to check for the existence of one or more required fields, eg the cpu value.

Reply:
Are there any APM nodes that don't produce `app`? `series.metricsets` includes every metric set attached to the node in question, so as long as `app` is present then the existence of other sets don't matter. 

I'm not sure how else to differentiate between a missing CPU value because of a No Data situation, and a missing CPU value because the node is from APM.